examples/bindgen_toolchain/MODULE.bazel

```diff
@@ -0,0 +1,16 @@
+module(name = "bindgen_toolchain_example")
+
+bazel_dep(name = "rules_rust_bindgen")
+local_path_override(module_name = "rules_rust_bindgen", path = "../../extensions/bindgen")
+
+bazel_dep(name = "bazel_skylib", version = "1.7.1")
+bazel_dep(name = "rules_cc", version = "0.0.17")
+bazel_dep(name = "rules_rust")
+local_path_override(module_name = "rules_rust", path = "../..")
+
+bazel_dep(name = "toolchains_llvm", version = "1.3.0")
+llvm = use_extension("@toolchains_llvm//toolchain/extensions:llvm.bzl", "llvm")
+llvm.toolchain(llvm_version = "17.0.6")
+use_repo(llvm, "llvm_toolchain_llvm")
+
+register_toolchains("//:my_bindgen_toolchain")
```

Comment:
This is caused because you haven't registered a `toolchain` target but instead a `rust_bindgen_toolchain` target. You're missing that intermediate.

Reply:
This I think is the solution to https://github.com/bazelbuild/rules_rust/pull/3231#issuecomment-2656629681

Reply:
great point! I missed that one somehow. Now it works 👍